src/iptc.cpp

```diff
@@ -79,6 +80,90 @@ namespace {
         uint16_t record_;
 
     }; // class FindIptcdatum
+
+    bool is_utf8(const std::string& string)
+    {
+        if(string.empty())
+            return 0;
+
+        size_t i = 0;
+        while(i < string.size())
+        {
+            unsigned char c1 = string[i];
+            if( (// ASCII
+                 // use bytes[i] <= 0x7F to allow ASCII control characters
+                    c1 == 0x09 ||
+                    c1 == 0x0A ||
+                    c1 == 0x0D ||
+                    (0x20 <= c1 && c1 <= 0x7E)
+                )
+            ) {
+                i += 1;
+                continue;
+            }
+
+            unsigned char c2 = string[i+1];
```

Comment:
This will cause an out of bounds read when `i` reaches `string.size() - 1` and the string is malformed (i.e. not NULL terminated).